src/commands/scaling/addScaleRule/AddScaleRuleStep.ts

```diff
@@ -3,38 +3,53 @@
 import { localize } from "../../../utils/localize";
-import { updateContainerApp } from "../../../utils/updateContainerApp";
+import { RevisionDraftUpdateBaseStep } from "../../revisionDraft/RevisionDraftUpdateBaseStep";
 import type { IAddScaleRuleContext } from "./IAddScaleRuleContext";
 
-export class AddScaleRuleStep extends AzureWizardExecuteStep<IAddScaleRuleContext> {
-    public priority: number = 100;
+export class AddScaleRuleStep<T extends IAddScaleRuleContext> extends RevisionDraftUpdateBaseStep<T> {
+    public priority: number = 200;
 
-    public async execute(context: IAddScaleRuleContext, _progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
-        const adding = localize('addingScaleRule', 'Adding {0} rule "{1}" to "{2}"...', context.ruleType, context.ruleName, context.containerApp.name);
-        const added = localize('addedScaleRule', 'Successfully added {0} rule "{1}" to "{2}".', context.ruleType, context.ruleName, context.containerApp.name);
+    constructor(baseItem: RevisionsItemModel) {
+        super(baseItem);
+    }
+
+    public async execute(context: IAddScaleRuleContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        let adding: string | undefined;
+        let added: string | undefined;
+        if (context.containerApp.revisionsMode === KnownActiveRevisionsMode.Single) {
+            adding = localize('addingScaleRuleSingle', 'Add {0} rule "{1}" to container app "{2}" (draft)', context.ruleType, context.ruleName, context.containerApp.name);
+            added = localize('addedScaleRuleSingle', 'Added {0} rule "{1}" to container app "{2}" (draft).', context.ruleType, context.ruleName, context.containerApp.name);
+        } else {
+            adding = localize('addingScaleRuleMultiple', 'Add {0} rule "{1}" to revision "{2}" (draft)', context.ruleType, context.ruleName, this.baseItem.revision.name);
+            added = localize('addedScaleRuleMultiple', 'Added {0} rule "{1}" to revision "{2}" (draft)', context.ruleType, context.ruleName, this.baseItem.revision.name);
+        }
 
-        const template: Template = context.containerApp.template || {};
-        template.scale = context.scale || {};
-        template.scale.rules = context.scaleRules || [];
+        context.activityTitle = adding;
+        progress.report({ message: localize('addingRule', 'Adding scale rule...') });
+
+        this.revisionDraftTemplate.scale ||= {};
+        this.revisionDraftTemplate.scale.rules ||= [];
 
         const scaleRule: ScaleRule = this.buildRule(context);
-        this.integrateRule(context, template.scale.rules, scaleRule);
+        this.integrateRule(context, this.revisionDraftTemplate.scale.rules, scaleRule);
+        this.updateRevisionDraftWithTemplate();
+
+        // Artificial delay to make the activity log look like it's performing an action
+        await delay(1000);
```

Comment:
I feel like a second is a little bit long to block a user for something that's artificial.

My understanding is that it's so that you can see `Adding scale rule...` appear, but even if they don't see that progress message, they can still see that the "Added scale rule to container app (draft)` completed successfully, right?

I don't know if we need that progress message since it's instantaneous to begin with.

Reply:
If we do want to keep the delay though, I think we should put the delay in `updateRevisionDraftWithTemplate` so that all of the actions behave similarily.